micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/ExecutorServiceMetrics.java

```diff
@@ -209,6 +210,96 @@ private void monitor(MeterRegistry registry, @Nullable ThreadPoolExecutor tp) {
                 .description("The current number of threads in the pool")
                 .baseUnit(BaseUnits.THREADS)
                 .register(registry);
+
+        monitorRejectedExecutionHandler(registry, tp);
+        monitorThreadFactory(registry, tp);
+    }
+
+    private void monitorThreadFactory(MeterRegistry registry, ThreadPoolExecutor tp) {
+        MonitoredThreadFactory tf = new MonitoredThreadFactory(tp.getThreadFactory());
```

Comment:
Probably good to check that we're not wrapping something already wrapped here. That is, if `tp.getThreadFactory` is already of type `MonitoredThreadFactory`